subsys/spm/CMakeLists.txt

```diff
@@ -5,3 +5,30 @@
 #
 
 zephyr_sources(spm.c)
+
+if (IMAGE)
+  # Only perform merge operation when being part of multi image build.
+  set_property(
+    GLOBAL APPEND PROPERTY
+    HEX_FILES_TO_MERGE
+    ${PROJECT_BINARY_DIR}/${KERNEL_HEX_NAME}
+
+    )
+  set_property(
+    GLOBAL APPEND PROPERTY
+    HEX_FILES_TO_MERGE_TARGET
+    ${IMAGE}kernel_elf
+    )
+
+  set_property(
+    GLOBAL APPEND PROPERTY
+    HEX_FILES_TO_MERGE
+    ${CMAKE_BINARY_DIR}/zephyr/${KERNEL_HEX_NAME}
+
+    )
+  set_property(
+    GLOBAL APPEND PROPERTY
+    HEX_FILES_TO_MERGE_TARGET
+    kernel_elf
+    )
```

Comment:
We are inconsistent with who modifies this property. In SPM it does done in a subsys. In MCUBoot it is done by the child image build scripts.

Reply:
I agree.

This a temporary solution until the final solution based on either partition manager or some other scheme is in place.

Reply:
Added to TODO list